src/libusb/device-libusb.cpp

```diff
@@ -0,0 +1,92 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#include "device-libusb.h"
+#include "endpoint-libusb.h"
+#include "interface-libusb.h"
+#include "types.h"
+
+#include <string>
+#include <regex>
+#include <sstream>
+#include <mutex>
+
+namespace librealsense
+{
+    namespace platform
+    {
+        std::string get_usb_descriptors(libusb_device* usb_device)
+        {
+            auto usb_bus = std::to_string(libusb_get_bus_number(usb_device));
+
+            // As per the USB 3.0 specs, the current maximum limit for the depth is 7.
+            const auto max_usb_depth = 8;
+            uint8_t usb_ports[max_usb_depth] = {};
+            std::stringstream port_path;
+            auto port_count = libusb_get_port_numbers(usb_device, usb_ports, max_usb_depth);
+            auto usb_dev = std::to_string(libusb_get_device_address(usb_device));
+            auto speed = libusb_get_device_speed(usb_device);
+            libusb_device_descriptor dev_desc;
+            auto r= libusb_get_device_descriptor(usb_device,&dev_desc);
+
+            for (size_t i = 0; i < port_count; ++i)
+            {
+                port_path << std::to_string(usb_ports[i]) << (((i+1) < port_count)?".":"");
+            }
+
+            return usb_bus + "-" + port_path.str() + "-" + usb_dev;
+        }
+
+        usb_device_libusb::usb_device_libusb(libusb_device* device, libusb_device_descriptor desc) :
```

Comment:
Can you pass the params by (const) ref?

Reply:
no, libusb api doe's not accept `const libusb_device*`, other params will be changed to const